feat: Switch currently displayed hyperplane

iluha168 opened the pull request.

iluha168 committed:
add keybinding and state for current perspective

iluha168 committed:
show current hyperplane in F3 menu

inayakitori commented:
Some goals for this are moving towards instead having a record that stores the chunk rending directions, which allows for positive and negative rotations:
```java
// possible values for chunkX and chunkZ are two orthogonal values from [X+, X-, Z+, Z-, W+, W-]
record Perspective4 (Direction chunkX, Direction chunkZ){
  

  Perspective4 rotateWToForward(Direction currentFacingChunkDirection, AxisDirection wDir){
    // rotates according to the currently facing chunk direction
    // if wDir is positive, the relative ana direction is rotated into the forward direction. otherwise, the relative kata direction is
  }
  
  Perspective4 rotateWToRight(Direction currentFacingChunkDirection, AxisDirection wDir){
    // rotates according to the currently facing chunk direction
    // if wDir is positive, the relative ana direction is rotated into the right direction. otherwise, the relative kata direction is
  }

 
}
```

Further goals are:
- to find a way to then use this Perspective4 to modify the blockstate on render
- Render the debug (F3) cursor so that it correctly displays the current rotation
- Find out how to apply transformations to the rendering so mirroring and other stuff renders correctly


Examples of debug cursors:

Perspective4(X+,Z+) (implicit third direction is W+)
<img width="172" height="137" alt="image" src="https://github.com/user-attachments/assets/a36f61fc-2f2d-4faf-b95c-11d6c22473d1" />

Perspective4(Z+,X+) (implicit third direction is W-) (this is a mirrored slice)
<img width="172" height="137" alt="image" src="https://github.com/user-attachments/assets/88f03aa0-4ead-4e59-8a83-bed5ed7ed298" />

Perspective4(Z+,W+) (implicit third direction is X+)
<img width="172" height="137" alt="image" src="https://github.com/user-attachments/assets/79b8228b-c6d4-4142-945a-38386530ac1a" />

iluha168 committed:
better rotation keybind logic

iluha168 committed:
accurate debug crosshair

inayakitori committed:
Implemented getting the block at the correct location after rotating the slice

iluha168 committed:
mv IDebugHudMixin

inayakitori committed:
Add config lang keys and make FixedDirection config option a button instead of a dropdown

iluha168 committed:
reload world on perspective4 shift

iluha168 committed:
fix: accurate `projectInverse`

iluha168 committed:
fix: `fdmc$followCurrent4DPerspective`

inayakitori committed:
Add shorthand hyperplane gui option

iluha168 committed:
first-person camera following Perspective4

iluha168 committed:
QoL: rotate camera along with perspective4

iluha168 commented:
Rebase on top of `develop`

iluha168 committed:
fixup! first-person camera following Perspective4

inayakitori committed:
Remap inputs to follow current rendering

iluha168 committed:
Revert "Remap inputs to follow current rendering"

This reverts commit 55b49e95bb462e151bbe70f6d031481b1124b2c9.

iluha168 committed:
rotate movementInput according to perspective4

iluha168 committed:
add changelog

iluha168 committed:
start crosshair target raycast at camera render position

inayakitori committed:
Moved the perspective4 from being a GameOption to being a TrackedData entry of any Entity
Created Perspective4 Codecs
Fixed Direction CODECS

iluha168 committed:
return clientside optimistic perspective4 change

iluha168 committed:
cleanup ClientEntityMixin because perspective4 is serverside now

iluha168 committed:
fix movement input vector length

iluha168 committed:
fix debug cursor perspective4 desync

iluha168 committed:
fixup! fix debug cursor perspective4 desync

inayakitori committed:
Moved ScreenHandler to remove registry actions in FDMCConstants
Unit tests set up
Unit tests for Perspective4 (more to come)
Github Test actions

inayakitori committed:
Moved Perspective4.VALUES to tests
added a couple more tests checking rotational identities.

iluha168 committed:
accurate pick block command

iluha168 committed:
somewhat accurate block breaking and placing

iluha168 committed:
fix: be able to run dedicated server

inayakitori committed:
movement changes and sprint fix

inayakitori committed:
singleplayer fix dedicated potentially broken

iluha168 committed:
cosmetic cleaning

inayakitori committed:
single WrapMethod lighting fix raaaaaaa

iluha168 committed:
debug hud: configurable color of 4th axis, render position display, better `Perspective4.toString`

iluha168 committed:
fixup! debug hud: configurable color of 4th axis, render position display, better `Perspective4.toString`

inayakitori committed:
Command to get, set, and reset Perspective4. More unit tests to come

inayakitori committed:
Perspective4ArgumentType suggestions and small fix

iluha168 committed:
reload world only once per perspective change
